storage/innobase/xtrabackup/src/xbcloud/xbcloud.cc

```diff
@@ -1242,6 +1234,33 @@ int main(int argc, char **argv) {
     container_name = opt_swift_container;
 
   } else if (opt_storage == S3) {
+    std::shared_ptr<S3_ec2_instance> ec2_instance =
```

Comment:
Better to create object if access_key, secret_key are not specified.

Reply:
We cannot do it. We cannot declare a var inside a loop and use it outside the loop. We will use ec2_instance later to pass ownership of the pointer to Object_store.